components/raven_tools/actions/add-keyword/add-keyword.ts

```diff
@@ -0,0 +1,46 @@
+import app from "../../app/raven_tools.app";
+import { defineAction } from "@pipedream/types";
+import { DOCS } from "../../common/constants";
+import { AddKeywordParams, RavenToolsResponse } from "../../common/types";
+
+export default defineAction({
+  name: "Add Keyword",
+  description: `Add a keyword to a domain [See docs here](${DOCS.addKeyword})`,
+  key: "raven_tools-add-keyword",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    app,
+    domain: {
+      propDefinition: [app, "domain"],
+      description: "The domain to add the keyword to.",
+    },
+    keyword: {
+      label: "Keyword",
+      description: "The keyword to add.",
+      type: "string",
+    },
+  },
+  async run({ $ }): Promise<RavenToolsResponse> {
+    const { domain, keyword } = this;
+    const params: AddKeywordParams = {
+      $,
+      params: {
+        domain,
+        keyword,
+      },
+    };
+    const data: RavenToolsResponse = await this.app.addKeyword(params);
+
+    if (data?.response === "success") {
+      $.export("$summary", `Successfully added keyword "${keyword}"`);
+      return data;
+    } else {
+      throw new Error(
+        `Something went wrong. RavenTools response: ${
+          data && JSON.stringify(data)
+        }`
+      );
+    }
```

Comment:
I wouldn't make it a ConfigurationError since this is a "catch-all" error based on the API response - they could return an error for any number of reasons unrelated to the user input.

But I agree with the structuring - the if-else was unnecessary, I'm simplifying it as per your suggestion. Thanks!

Reply:
I wouldn't make it a ConfigurationError since this is a "catch-all" error based on the API response - they could return an error for any number of reasons unrelated to the user input.

But I agree with the structuring - the if-else was unnecessary, I'm simplifying it as per your suggestion. Thanks!